fairlearn/grid_search/binary_protected_attribute/binary_classification.py

```diff
@@ -0,0 +1,123 @@
+    accuracy
+
+    :param learner: An object which can be used to fit a model to
+    features, labels and weights. A deep copy of this is made for each
+    value of the lagrange multiplier used
+    :type learner: Must implement a fit(x, y, sample_weight) method
+
+    :param x: The array of training data features (which may or may
+    not contain the protected attribute). Must have as many rows
+    as y and protected_attribute
+    :type x: Nested list or numpy array with two dimensions or
+    pandas dataframe
+
+    :param y: The list of binary classes, which must be 0 or 1. Must
+    contain the same number of entries as rows in x. We do not
+    consider it an error if only one class is present
+    :type y: List or Numpy array with one dimension
+
+    :param protected_attribute: The binary protected attribute
+    corresponding to each row of x. We do not consider it an error
+    if only one attribute value is present
+    :type protected_attribute: List or Numpy array with one dimension
+
+    :param lagrange_multipliers: User specified set of Lagrange
+    multipliers to use for the optimisation problem. If this is set
+    then number_lagrange_multipliers must be None.
+    The result array will be equal in length to this array
+    :type lagrange_multipliers: List of real numbers
+
+    :param number_lagrange_multipliers: Specifies the number of
+    Lagrange multipliers to use in the optimisation problem. If this
+    is set then lagrange_multipliers must be None.
+    The result array will have as many entries as specified here
+
+    :return: The models corresponding to each value of the Lagrange
+    multiplier tested
+    :rtype: List of dictionaries. Each dictionary has fields
+    "lagrange_multiplier" and "model." Each model will correspond to
+    the input parameter learner after calling 'fit' on it (a deep
+    copy is made). The user is responsible for converting these
+    objects to an actual model, if further processing is required.
+    """
+    # Must specify either an array of Lagrange multipliers or how many
+    # of them to generate
+    if bool(lagrange_multipliers is None) == bool(number_lagrange_multipliers is None):  # noqa: E501
+        raise RuntimeError("Must specify either lagrange_multipliers "
+                           "or number_lagrange_multipliers")
+
+    # Verify we have a binary classification problem
+    unique_labels = np.unique(y)
+    if len(unique_labels) > 2:
+        raise RuntimeError("Supplied Y labels are not binary")
+
+    if not set(unique_labels).issubset({0, 1}):
+        raise RuntimeError("Supplied Y labels are not 0 or 1")
+
+    # Extract required statistics from protected_attribute
+    p0, p1, a0_val = utilities.generate_protected_attribute_info(
+        protected_attribute)
+
+    # If not supplied, generate array of trial lagrange multipliers
+    if lagrange_multipliers is None:
+        limit = 1
+        if p0 / p1 > 1:
+            limit = p0 / p1
```

Comment:
division by zero; when p1=0, this will be infinity

Reply:
Yes, this is similar to the case where all the Y labels are the same. In that case we press on and return a bunch of trivial learners. In this one... eventually an exception gets thrown. The ratio is infinity, and then np.linspace generates an array of NaNs.

I'll throw an exception in this case, since the failure isn't pretty